gen_data/gen_test_m24_single.py

```diff
@@ -0,0 +1,475 @@
+    for n in range(0, novelty_timestamp):
+        # before novelty_timestamp only sample from known_videos
+        vid_id = np.random.choice(len(known_videos))
+        vid = known_videos[vid_id]
+        df.loc[len(df.index)] = [vid, 0, 0, known_videos_dict[vid], 0, 0]
+        if known_videos_dict[vid] not in known_classes:
+            known_classes.append(known_videos_dict[vid])
+        del known_videos[vid_id]
+
+    if aug_type == "spatial":
+        is_spatial = 1
+        is_temporal = 0
+    elif aug_type == "temporal":
+        is_spatial = 0
+        is_temporal = 1
+    elif aug_type in ['class','perspective','ontology_id',
+            'location','relation','relation_type']:
+        is_spatial = is_temporal = 0
+    else:
+        raise NotImplementedError('Unknown aug_type {}'.format(aug_type))
+
+    combine_known_unknown = [(k, 'known') for k in known_videos]
+    combine_known_unknown += [(k, 'unknown') for k in unknown_videos]
+    
+    red_light, red_light_det = None, -1
+    for n in range(novelty_timestamp, num_total_samples):
+        vid_id = np.random.choice(len(combine_known_unknown))
+        vid, novelty_str = combine_known_unknown[vid_id]
+        if novelty_str == 'known':
+            is_novel = 0
+            activity = known_videos_dict[vid]
+            if activity not in known_classes:
+                known_classes.append(activity)
+        elif novelty_str == 'unknown':
+            is_novel = 1
+            activity = unknown_videos_dict[vid]
+            if activity not in unknown_classes:
+                unknown_classes.append(activity)
+            
+            if make_novel_all_one_class:
+                # Make all unknown activities single class
+                assert 'novel' not in known_classes
+                activity = 'novel'
+
+            # TODO: decide what to do with red_light when at novelty_timestamp
+            # but sampling from known
+            if red_light is None:
+                red_light = vid
+                red_light_det = n
+        else:
+            raise ValueError(f'Unkown novelty_str: {novelty_str}')
+
+        df.loc[len(df.index)] = [
+            vid,
+            is_novel,
+            1,
+            activity,
+            is_novel & is_spatial,
+            is_novel & is_temporal,
+        ]
+
+        del combine_known_unknown[vid_id]
+
+    assert len(combine_known_unknown) == 0
+  
+    metadata = {
+        "protocol": protocol,
+        "num_total_samples": num_total_samples,
+        "round_size": round_size,
+        "difficulty": None,  # figure out what this should be
+        "distribution": None,  # figure out what this should be
+        "n_rounds": math.ceil(
+            num_total_samples / round_size
+        ),  # TODO: confirm that this should be ceil
+        "representation": None,  # figure out what this should be
+        "threshold": 0.5,  # TODO: don't hardcode this
+        "pre_novelty_batches": math.ceil(
```

Comment:
I think this should be floor, since post novelty would be in the batch where the `red_light_det` is